drivers/i2s/i2s_ll_stm32.h

```diff
@@ -41,6 +41,9 @@ struct stream {
 	bool src_addr_increment;
 	bool dst_addr_increment;
 	uint8_t fifo_threshold;
+	uint8_t sample_tx_block_sent;
+	uint8_t sample_tx_block_dma_sent;
+	bool sample_tx_stop_drain;
```

Comment:
Nit: variable names are confusing, maybe names such as the following could be more appropriate?
* `uint8_t blocks_queued_by_api`
* `uint8_t blocks_sent_by_dma`
* `bool tx_stop_for_drain`